libraries/AP_Mount/AP_Mount_Tusuav.h

```diff
@@ -0,0 +1,414 @@
+
+    // // init - performs any required initialisation for this instance
+    // void init() override;
+
+    // update mount position - should be called periodically
+    void update() override;
+
+    // return true if healthy
+    bool healthy() const override;
+
+    // return true if this mount accepts roll targets
+    bool has_roll_control() const override { return false; };
+
+    // has_pan_control - returns true if this mount can control its pan (required for multicopters)
+    bool has_pan_control() const override { return true; };
+	
+    // take a picture.  returns true on success
+    bool take_picture() override;
+
+    // start or stop video recording
+    // set start_recording = true to start record, false to stop recording
+    bool record_video(bool start_recording) override;
+
+    // set zoom specified as a rate or percentage
+    bool set_zoom(ZoomType zoom_type, float zoom_value) override;
+
+    // set tracking to none, point or rectangle (see TrackingType enum)
+    // if POINT only p1 is used, if RECTANGLE then p1 is top-left, p2 is bottom-right
+    // p1,p2 are in range 0 to 1.  0 is left or top, 1 is right or bottom
+    bool set_tracking(TrackingType tracking_type, const Vector2f& p1, const Vector2f& p2) override;
+
+    // set camera lens as a value from 0 to 5
+    bool set_lens(uint8_t lens) override;
+
+    // set_camera_source is functionally the same as set_lens except primary and secondary lenses are specified by type
+    // primary and secondary sources use the AP_Camera::CameraSource enum cast to uint8_t
+    bool set_camera_source(uint8_t primary_source, uint8_t secondary_source) override;
+
+    // send camera information message to GCS
+    void send_camera_information(mavlink_channel_t chan) const override;
+
+    // send camera settings message to GCS
+    void send_camera_settings(mavlink_channel_t chan) const override;
+
+    // get rangefinder distance.  returns true on success
+    bool get_rangefinder_distance(float& distance_m) const override;
+
+    // enable/disable rangefinder.  returns true on success
+    bool set_rangefinder_enable(bool enable) override;
+
+protected:
+
+    // get attitude as a quaternion.  returns true on success
+    bool get_attitude_quaternion(Quaternion& att_quat) override;
+private:
+
+    // send text prefix string to reduce flash cost
+    static const char* send_text_prefix;
+
+    // gimbal main commands for controlling gimbal behavior
+    enum class GimbalCmd : uint8_t {
+        MasCmd_Load   = 0xBC,  // load settings or configurations
+        MasCmd_Gimbal = 0xBD,  // gimbal operations
+        MasCmd_Device = 0x0B   // device-specific operations
+    };
+
+    // gimbal sub-commands for controlling and retrieving gimbal status
+    enum class GimbalSubCmd : uint8_t {
+        GimbalSubCmd_GimbalAngle           =   0x16,   // retrieve gimbal angles
+        GimbalSubCmd_PlaneSensor_Attitude  =   0x18,   // send plane sensor & attitude to gimbal
```

Comment:
Seems to be some oddity in naming here.

Perhaps this enum should just be `Cmd`?

Or strip some component out of the entries?

Reply:
Thank you for your comments! The GimbalSubCmd enum represents sub-commands linked to MasCmd_Gimbal = 0xBD in the GimbalCmd enum. It defines specific operations for controlling and retrieving gimbal status, including angle retrieval, sensor attitude synchronization, parameter adjustments, laser measurement retrieval, and zoom or tracking configurations. We have reviewed the GimbalSubCmd entries and confirmed that each one is essential for its respective function.